applications/app/controllers/InteractiveController.scala

```diff
@@ -5,21 +5,42 @@ import common._
 import contentapi.ContentApiClient
 import conf._
 import conf.switches.Switches
+import model.Cached.{WithoutRevalidationResult, RevalidatableResult}
 import model._
+import play.api.libs.iteratee.Enumerator
+import play.api.libs.ws.{WSResponse, WSClient}
 import play.api.mvc._
 import views.support.RenderOtherStatus
+import conf.Configuration.interactive.cdnPath
+import conf.Configuration.environment.isPreview
+import scala.concurrent.duration._
 
 import scala.concurrent.Future
 
 case class InteractivePage (interactive: Interactive, related: RelatedContent) extends ContentPage {
   override lazy val item = interactive
 }
 
-class InteractiveController(contentApiClient: ContentApiClient) extends Controller with RendersItemResponse with Logging with ExecutionContexts {
+class InteractiveController(contentApiClient: ContentApiClient, wsClient: WSClient) extends Controller with RendersItemResponse with Logging with ExecutionContexts {
 
   def renderInteractiveJson(path: String): Action[AnyContent] = renderInteractive(path)
   def renderInteractive(path: String): Action[AnyContent] = Action.async { implicit request => renderItem(path) }
 
+  def proxyInteractiveServiceworker(path: String): Action[AnyContent] = Action.async { implicit request =>
+    val stage = if (isPreview) "preview" else "live"
+    val serviceWorkerPath = s"$cdnPath/service-workers/$stage/$path/interactive-service-worker.js"
+
+    wsClient.url(serviceWorkerPath).get().map { response =>
+      Cached (365.days) {
+        response.status match {
+          case 200 =>
+            RevalidatableResult(Ok(response.body).as("text/javascript; charset=utf8"), response.body)
```

Comment:
Would it make sense to use `response.allHeaders("Content-Type")` rather the hardcoding it?